### ISSUE TITLE
[feature request] turn back old Chanka plugins

### ISSUE THREAD

**Issue by GoogleCodeExporter** [aloone/nulldc#154]
```
Even if no more changes in it, this plugin is still worth for its low system 
requirements. Provide good speed and good graphics, turn it back please
```

Original issue reported on code.google.com by `old.liq...@gmail.com` on 27 Jun 2010 at 2:03

**Comment by GoogleCodeExporter** [comment]
```
Not possible due to licensing issues. Same goes for the other old plugins 
ported from Chankast. Not gonna happen.
```

Original comment by `Mr.PsyMan` on 27 Jun 2010 at 2:58
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
Chankst is licensed?? o_O
```

Original comment by `old.liq...@gmail.com` on 27 Jun 2010 at 2:27

**Comment by GoogleCodeExporter** [comment]
```
No, but nullDC now is... and this would create issues. I'm not going to try and 
explain any further. Let's just leave it there.
```

Original comment by `Mr.PsyMan` on 27 Jun 2010 at 2:51

**Comment by GoogleCodeExporter** [comment]
```
Very bad! VARY BAD! Becouse in some games Chanka plugin is MUCH BETTER! Many 
people think like that! MANY PEOPLE!
hell......
```

Original comment by `N2008eo...@rambler.ru` on 27 Jun 2010 at 3:37

**Comment by GoogleCodeExporter** [comment]
```
Why don't we try to improve the existing plugins to be even better than the 
Chankast plugins. By doing that, these plugins would be a reference and there 
would be no need for other plugins, such as the Chankast.
```

Original comment by `telmo.qu...@gmail.com` on 27 Jun 2010 at 4:11

**Comment by GoogleCodeExporter** [comment]
```
do you know how?
```

Original comment by `N2008eo...@rambler.ru` on 27 Jun 2010 at 4:44

**Comment by GoogleCodeExporter** [comment]
```
maybe adding emulation of not only standard functions, but other too, like 
sprite manipulation (seen in Echelon Skies of Arcadia cracktro, Chankast 
emulating it normally). Or add option to reduce the system requirements 
(disabling shaders or something like this). For DC absolutely normal machine is 
P4-2500 or 3000. DX10? What is DX10 for DC? Nothing.
```

Original comment by `old.liq...@gmail.com` on 3 Jul 2010 at 5:36

**Comment by GoogleCodeExporter** [comment]
```
"Old" rendering methods (fixed pipeline) are slower compared to shaders.
Not to mention that video cards using fixed pipeline should support quite some 
more extensions (ie: Intel chips do not) for the plugin to even work.
There is a reason why the Chankast plugin shows completely broken graphics on 
Intel cards.

Even nullPVR, which uses primarily shaders and has fixed pipeline as a last 
resort, does some hacky approaches on some cases cause of Direct3D 9.0c 
limitations.
There are even cases where nullPVR supports additional Direct3D 9.0c features 
that no graphics vendors support any more (ie: the D24FS8 Z-Buffer mode, which 
is the most accurate mode, was only supported on some ATi cards using Direct3D 
9.0c, then ATi dropped support for it, NVIDIA never supported it either as far 
as I remember).

Pentium 4 CPUs have slow SSE instructions. That is not so serious on other 
applications, but when emulating the DC it is a major issue. It makes the 
recompiler quite inefficient.

In short, a Pentium 4 is too slow for DC emulation, no matter how you look at 
it, excluding some very limited cases of "simple" games. A video card that does 
not support DirectX 10 (or newer) is gonna have graphical issues. Unfortunately 
nothing can be done about both of the problems above, other than using a decent 
CPU, a decent GPU and a DirectX 10 (or 11) video plugin.

The only workaround (not an actual fix) would be to hack things up in order to 
support a very specific game (fast without graphical or sound issues) while 
breaking everything else in the process. That will work if you really wanna 
play one game but for the sake of compatibility, consistency, etc, it's not 
gonna happen.
```

Original comment by `Mr.PsyMan` on 3 Jul 2010 at 9:50
